include/eve/module/core/regular/impl/simd/riscv/simd_cast.hpp

```diff
@@ -0,0 +1,183 @@
+    else if constexpr( bit_size == 2 ) return __riscv_vreinterpret_b2(x);
+    else if constexpr( bit_size == 4 ) return __riscv_vreinterpret_b4(x);
+    else if constexpr( bit_size == 8 ) return __riscv_vreinterpret_b8(x);
+    else if constexpr( bit_size == 16 ) return __riscv_vreinterpret_b16(x);
+    else if constexpr( bit_size == 32 ) return __riscv_vreinterpret_b32(x);
+    else if constexpr( bit_size == 64 ) return __riscv_vreinterpret_b64(x);
+  }
+}
+
+template<scalar_value T, typename N, scalar_value U, typename M>
+EVE_FORCEINLINE wide<U, M>
+                rvv_simd_cast(logical<wide<T, N>> x, as<wide<U, M>> const&) noexcept
+requires rvv_abi<abi_t<T, N>>
+{
+  constexpr auto out_lmul = rvv_lmul_v<U, M>;
+  static_assert(out_lmul == 1,
+                "[riscv eve] Can not bitcast from logical not to m1 vector register");
+  constexpr auto c = categorize<wide<U, M>>();
+  if constexpr( match(c, category::int64) ) return __riscv_vreinterpret_i64m1(x);
+  else if constexpr( match(c, category::uint64) ) return __riscv_vreinterpret_u64m1(x);
+  else if constexpr( match(c, category::int32) ) return __riscv_vreinterpret_i32m1(x);
+  else if constexpr( match(c, category::uint32) ) return __riscv_vreinterpret_u32m1(x);
+  else if constexpr( match(c, category::int16) ) return __riscv_vreinterpret_i16m1(x);
+  else if constexpr( match(c, category::uint16) ) return __riscv_vreinterpret_u16m1(x);
+  else if constexpr( match(c, category::int8) ) return __riscv_vreinterpret_i8m1(x);
+  else if constexpr( match(c, category::uint8) ) return __riscv_vreinterpret_u8m1(x);
+  else if constexpr( match(c, category::float32) )
+    return __riscv_vreinterpret_f32m1(__riscv_vreinterpret_u32m1(x));
+  else if constexpr( match(c, category::float64) )
+    return __riscv_vreinterpret_f64m1(__riscv_vreinterpret_u64m1(x));
+}
+
+// logical-logical - with logical-wide-logical.
+// it is expected that it could be called with different N/M
+// * N >= M: when we have is_full false, we first work with full_type, that bitcast it to smaller
+// type.
+// * N < M: is_full false, we could cast from not full type to full one.
+template<callable_options O, scalar_value T, typename N, scalar_value U, typename M>
+EVE_FORCEINLINE logical<wide<U, M>>
+                simd_cast_(EVE_REQUIRES(rvv_),
+                           const O&,
+                           logical<wide<T, N>>            x,
+                           as<logical<wide<U, M>>> const                &tgt) noexcept
+requires(rvv_abi<abi_t<T, N>> || rvv_abi<abi_t<U, M>>)
+{
+  if constexpr( is_aggregated_v<abi_t<T, N>> || is_aggregated_v<abi_t<U, M>> )
+  {
+    auto [lv, hv] = x.slice();
+    auto half_tgt = as<logical<wide<U, typename M::split_type>>> {};
+    auto to_ret   = logical<wide<U, M>> {simd_cast(lv, half_tgt), simd_cast(hv, half_tgt)};
+    return to_ret;
+  }
+  else
+  {
+    using u_m1_cardinal  = typename riscv_::m1_fixed<std::uint32_t>;
```

Comment:
You can probably have `using rvv_m1_wide = ` somewhere.

Reply:
Done. Moved this to rvv_utils